src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -330,12 +335,27 @@ private boolean checkImmutableIndices(Object request, ActionListener listener) {
         }
         
         if(request instanceof IndexRequest) {
-            if(complianceConfig != null && complianceConfig.isIndexImmutable(request)) {
+            if (isIndexImmutable(request)) {
                 ((IndexRequest) request).opType(OpType.CREATE);
             }
         }
         
         return false;
     }
 
-}
\ No newline at end of file
+    private boolean isIndexImmutable(Object request) {
+        if (complianceConfig == null || !complianceConfig.isEnabled()) {
```

Comment:
Actually this check is redundant. There is the same condition on line 198 before `checkImmutableIndices()` call.

Reply:
Well, I think this should be retained here while calling this function

Reply:
Why? It is a private method and there is no need to duplicate the check.